### DISCUSSION TITLE
File upload and download

### DISCUSSION THREAD

**Discussion by IvanDrag0** [lowdefy/lowdefy#1681] [Questions and Help]
Does lowdefy have the ability to upload and download files? I found an old [topic](https://github.com/lowdefy/lowdefy/discussions/921) that mentions adding a file selector in v4, however, I can't seem to find it in the documentation. Has this feature been implemented?

In the same note, does lowdefy allow the sending and receiving of binary files (e.g. using a file upload dialog to store/send data)?

**Comment by SamTolmay** [comment]
Hi @IvanDrag0 

Unfortunately we have not yet implemented a way to upload and download files easily, but this is something that is on our roadmap.

For now, you could create a custom block plugin, based on the code shared in the post you linked and/or the implementation of the `S3UploadButton` block.

Custom plugin documentation:
https://docs.lowdefy.com/plugins-introduction
https://github.com/lowdefy/lowdefy-example-plugins


S3UploadButton block code:
https://github.com/lowdefy/lowdefy/tree/main/packages/plugins/plugins/plugin-aws/src/blocks/S3UploadButton
